draft-yasskin-http-origin-signed-responses.md

```diff
@@ -425,24 +437,21 @@ there are no non-significant response header fields in the exchange.
          1. The text string "certSha256" to the byte string `certSha256`.
       1. The text string "date" to the integer value of `date`.
       1. The text string "expires" to the integer value of `expires`.
-      1. The text string "exchange" to the CBOR representation
-         ({{cbor-representation}}) of `exchange`. See the [open
-         questions](#incremental-validity).
+      1. The text string "headers" to the CBOR representation
+         ({{cbor-representation}}) of `exchange`'s headers.
 1. If `signature` is `message`'s signature by `main-certificate`'s public key
    using `signing-alg`, return "potentially-valid" with `exchange` and whichever
    is present of `certificate-chain` or `ed25519Key`. Otherwise, return
    "invalid".
 
-### Open Questions
-
-Including the entire exchange in the signed data forces a client to download the
-whole thing before trusting any of it. {{?I-D.thomson-http-mice}} is designed to
-let us check the validity of just the `MI` header up front and then
-incrementally check blocks of the payload as they arrive. What's the best way to
-integrate that? Maybe add a flag to the `Signature` header field or its
-signatures saying that the payload is guarded by some other header field, so
-isn't included in the significant parts ({{significant-parts}}).
-{:#incremental-validity}
+   Note that the above algorithm can determine that an exchange's headers are
+   potentially-valid before the exchange's payload is received. The client MAY
+   process those headers as soon as they are validated. If `integrity`
```

Comment:
> as soon as they are validated
Would you clarify if this means "potentially-validated" from Section 3.6 algorithm, or "valid" from the full cert chain authority check in Section 4.3.1.

Reply:
This turns out to be hard to word. It's actually both: there's a stack of standards here, and the next layer up mustn't process past the potentially-valid point when determining actual validity, and the layer above that mustn't process past the actually-valid point when using the data for some other purpose. It's totally plausible that a protocol might need to see the end of a message body to determine whether any of it is valid, but still take advantage of the incremental validity we're providing here to give that answer faster.

I've updated the wording to try to capture that. I'm going to merge this to master to make my next PR easier to review, but please let me know if I need to change it more.